src/main/java/org/eln2/mc/common/events/EventScheduler.kt

```diff
@@ -0,0 +1,93 @@
+package org.eln2.mc.common.events
+
+import net.minecraftforge.event.TickEvent
+import net.minecraftforge.eventbus.api.SubscribeEvent
+import net.minecraftforge.fml.common.Mod
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/**
+ * The event scheduler can be used to schedule events from any thread, that are to be sent on a game tick.
```

Comment:
> that are to be sent on a game tick.
Do you mean processed by a game tick?

Can this be used in contexts outside of running things on a game tick, if desired?